error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/DirectReturnTest.java

```diff
@@ -0,0 +1,174 @@
+            "  }",
+            "",
+            "  Object salientMockAssignmentToParam(String param) {",
+            "    param = mock();",
+            "    return param;",
+            "  }",
+            "",
+            "  String redundantAssignmentToLocalVariable() {",
+            "    String variable = null;",
+            "    // BUG: Diagnostic contains:",
+            "    variable = toString();",
+            "    return variable;",
+            "  }",
+            "",
+            "  String unusedAssignmentToLocalVariable(String param) {",
+            "    String variable = null;",
+            "    variable = toString();",
+            "    return param;",
+            "  }",
+            "",
+            "  String redundantVariableDeclaration() {",
+            "    // BUG: Diagnostic contains:",
+            "    String variable = toString();",
+            "    return variable;",
+            "  }",
+            "",
+            "  String redundantSpyVariableDeclaration() {",
+            "    // BUG: Diagnostic contains:",
+            "    String variable = spy();",
+            "    return variable;",
+            "  }",
+            "",
+            "  Object redundantSpyWithExplicitTypeVariableDeclaration() {",
+            "    // BUG: Diagnostic contains:",
+            "    String variable = spy(String.class);",
+            "    return variable;",
+            "  }",
+            "",
+            "  Object salientSpyTypeVariableDeclaration() {",
+            "    String variable = spy(\"name\");",
+            "    return variable;",
+            "  }",
+            "",
+            "  String unusedVariableDeclaration(String param) {",
+            "    String variable = toString();",
+            "    return param;",
+            "  }",
+            "",
+            "  String assignmentToAnnotatedVariable() {",
+            "    @SuppressWarnings(\"HereBeDragons\")",
+            "    String variable = toString();",
+            "    return variable;",
+            "  }",
+            "",
+            "  String complexReturnStatement() {",
+            "    String variable = toString();",
+            "    return variable + toString();",
+            "  }",
+            "",
+            "  String assignmentInsideIfClause() {",
+            "    String variable = null;",
+            "    if (true) {",
+            "      variable = toString();",
+            "    }",
+            "    return variable;",
+            "  }",
+            "",
+            "  String redundantAssignmentInsideElseClause() {",
+            "    String variable = toString();",
+            "    if (true) {",
+            "      return variable;",
+            "    } else {",
+            "      // BUG: Diagnostic contains:",
+            "      variable = \"foo\";",
+            "      return variable;",
+            "    }",
+            "  }",
+            "",
+            "  Supplier<String> redundantAssignmentInLambda() {",
```

Comment:
```suggestion
            "  Supplier<String> redundantAssignmentInsideLambda() {",
```
Also matches the method above.